### ISSUE TITLE
Custom semantic action doesn't work on iOS when checked property is not null

### ISSUE THREAD

**Issue by Haquoctien** [flutter/flutter#89728]
Hi, I'm having trouble with custom action on iOS VoiceOver. Custom actions won't show up in the [rotor menu](https://support.apple.com/en-us/HT204783) when `checked` property of Semantics is not `null`. This is only an issue for iOS in my case, on Android the action shows up in the local context menu just fine.
Code snippet:
```dart
Semantics(
  checked: true,
  customSemanticsActions: {
    CustomSemanticsAction(label: "Example action"): () {},
  },
  child: Container(
    height: 30,
    width: 30,
    color: Colors.amber,
  ),
),
```
Flutter doctor:
```
[✓] Flutter (Channel stable, 2.2.2, on macOS 11.5.2 20G95 darwin-x64, locale en-VN)
    • Flutter version 2.2.2 at /Users/haquoctien/development/flutter
    • Framework revision d79295af24 (3 months ago), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/haquoctien/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/haquoctien/Library/Android/sdk
    • Java binary at: /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (build 11.0.11+9)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (2 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 93.0.4577.63

! Doctor found issues in 1 category.
```

**Comment by darshankawar** [comment]
@Haquoctien 
Tried your code sample on latest stable (2.5.0) and below's my observation:

1. When `checked` is `null`:

voiceover reads, actions available: no voiceover.
double tapping on container, no voiceover.

2. When `checked` is `true`:

voiceover reads, actions available: no voiceover,
but double tapping on container, it reads `ON`

Is this a correct interpretation of the issue ?

What's the expected behavior in this case ?

**Comment by Haquoctien** [comment]
> @Haquoctien
> Tried your code sample on latest stable (2.5.0) and below's my observation:
> 
> 1. When `checked` is `null`:
> 
> voiceover reads, actions available: no voiceover.
> double tapping on container, no voiceover.
> 
> 1. When `checked` is `true`:
> 
> voiceover reads, actions available: no voiceover,
> but double tapping on container, it reads `ON`
> 
> Is this a correct interpretation of the issue ?
> 
> What's the expected behavior in this case ?

Hi, thanks for responding.
The expected behaviour would be when you use the rotor (by touching the screen with 2 fingers and rotate), you would see a a section in the rotor called "Actions", and you select that option by lifting your finger, now you can hear action label: Example action, and then you can carry out the action by double tap the widget.
The observed behavior is that, when checked set to non-null, there is no "Action" section in the rotor.

**Comment by darshankawar** [comment]
@Haquoctien
Thanks for the additional details. I re-verified using that info and was able to see `Action` section in the rotor when `checked: true`, ie, that section is shown using non-null `checked` property. Below is the screenshot of it:

<img width="352" alt="Screenshot 2021-09-09 at 6 35 23 PM" src="https://user-images.githubusercontent.com/67046386/132691338-c21e51fa-3e1f-49e3-ae68-3cc2cdc68f83.png">

Furthermore, when I select `Action` section and swipe up, voiceover reads `example action` correctly as expected.

I verified this on latest stable.

 <details>
<summary> flutter doctor -v </summary>

```
[✓] Flutter (Channel stable, 2.5.0, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
    • Flutter version 2.5.0 at /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4cc385b4b8 (31 hours ago), 2021-09-07 23:01:49 -0700
    • Engine revision f0826da7ef
    • Dart version 2.14.0

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.58.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 93.0.4577.63

• No issues found!

```

</details>

Maybe once you select any section and swipe up or down, that section goes away from rotor and hence you may not been seeing `Action` section in it.

**Comment by Haquoctien** [comment]
Hi, I tried upgrading to Flutter 2.5 and it worked!
So the issue must have been fixed somewhere between 2.2 and 2.5.
Thanks a lot :)))
